tools/metrics/gauges.go

```diff
@@ -4,13 +4,48 @@ import (
 	"fmt"
 	"log"
 	"net/url"
+	"strings"
 	"time"
 
 	"code.cloudfoundry.org/lager"
 	"github.com/alphagov/paas-cf/tools/metrics/pingdumb"
+	"github.com/alphagov/paas-cf/tools/metrics/tlscheck"
 	cfclient "github.com/cloudfoundry-community/go-cfclient"
 )
 
+func TLSValidityGauge(logger lager.Logger, addr string, interval time.Duration) MetricReadCloser {
+	return NewMetricPoller(interval, func(w MetricWriter) error {
+		if !strings.Contains(addr, ":") {
+			addr += ":443"
```

Comment:
What happens if mis-configuration results in the `addr` being `https://example.com:9000`?

Reply:
🙌  I had hoped that by naming the configuration variable `TLS_DOMAINS` rather than `_ENDPOINT` or `_URL` this would be avoided.

... I have added an additional check (by attempting to parse with `net.SplitHostAddr`) to catch this kind of misconfiguration a little better.